libraries/AP_GPS/AP_GPS_SBP2.cpp

```diff
@@ -476,7 +481,35 @@ AP_GPS_SBP2::logging_log_raw_sbp(uint16_t msg_type,
             time_us         : time_us,
             msg_type        : msg_type,
         };
-        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,192));
+        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,104));
         gps._DataFlash->WriteBlock(&pkt2, sizeof(pkt2));
+
+        if (msg_len > 168) {
+            struct log_SbpRAW2 pkt3 = {
+                LOG_PACKET_HEADER_INIT(LOG_MSG_SBPRAW2),
```

Comment:
SBR2 is a complementary message for SBR1, so the logic is to have as many SBR2 messages as needed to log the entire binary message. For the moment, the maximum length of an SBP message is 255, so we need 1 SBR1 and 2 SBR2 message maximum.